lua/telescope/builtin/lsp.lua

```diff
@@ -131,7 +131,7 @@ lsp.document_symbols = function(opts)
     return
   end
 
-  opts.ignore_filename = opts.ignore_filename or true
+  opts.ignore_filename = utils.get_default(opts.ignore_filename, true)
```

Comment:
I'll be honest: the way these options work is a bit wack. Basically you have two different options (`show_line` and `ignore_filename`), but only one of these can be enabled at a time. I decided not to refactor that as I'm not familiar enough with the Telescope code for that. Ideally this option is also renamed to `show_filename` with the default being `false`, as inverted options (`ignore_x` defaulting to `true`) are a bit harder to wrap your head around.

Reply:
Yeah i agree, but we kinda have this problem of changing option names that it would break things.
It happened and we have this option for 11 different pickers which makes it annoying.

`vim.F.if_nil` over `utils.get_default`. we wanted to move all of these over to `vim.F.if_nil` but havent found the time.

About options being a little bit wack i also agree. I think we did a way better job implementing `make_entry` for ctags. where you can have `show_line` and `filepath` which is there solely handled with `path_display`. So you can hide it with `path_display = { "hidden" }` maybe we should do this. for all other pickers (like the lsp once)